client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-recipe.ts

```diff
@@ -0,0 +1,130 @@
+import { useColorPaletteVariations, useFontPairingVariations } from '@automattic/global-styles';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState, useEffect, useRef, useCallback } from 'react';
+import { ONBOARD_STORE } from '../../../../../stores';
+import type { Pattern } from '../types';
+import type { OnboardSelect } from '@automattic/data-stores';
+import type { Design } from '@automattic/design-picker/src/types';
+import type { GlobalStylesObject } from '@automattic/global-styles';
+
+const useRecipe = ( siteId = 0, patterns: Pattern[] ) => {
+	const incrementIndexRef = useRef( 0 );
+	const [ header, setHeader ] = useState< Pattern | null >( null );
+	const [ footer, setFooter ] = useState< Pattern | null >( null );
+	const [ sections, setSections ] = useState< Pattern[] >( [] );
+	const [ colorVariation, setColorVariation ] = useState< GlobalStylesObject | null >( null );
+	const [ fontVariation, setFontVariation ] = useState< GlobalStylesObject | null >( null );
+	const selectedDesign = useSelect(
+		( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getSelectedDesign(),
+		[]
+	);
+
+	const {
+		stylesheet = '',
+		header_pattern_ids = [],
+		footer_pattern_ids = [],
+		pattern_ids = [],
+		color_variation_title = '',
+		font_variation_title = '',
+	} = selectedDesign?.recipe || {};
+
+	const colorVariations = useColorPaletteVariations( siteId, stylesheet );
+
+	const fontVariations = useFontPairingVariations( siteId, stylesheet );
+
+	const { setSelectedDesign } = useDispatch( ONBOARD_STORE );
+
+	const selectedDesignRef = useRef< Design | undefined >( selectedDesign );
+
+	selectedDesignRef.current = {
+		...selectedDesign,
+		recipe: {
+			...selectedDesign?.recipe,
+			header_pattern_ids: header ? [ header.id ] : [],
+			footer_pattern_ids: footer ? [ footer.id ] : [],
+			pattern_ids: sections.map( ( section ) => section.id ),
+			color_variation_title: colorVariation?.title,
+			font_variation_title: fontVariation?.title,
+		},
+	} as Design;
+
+	const snapshotRecipe = useCallback( () => setSelectedDesign( selectedDesignRef.current ), [] );
+
+	/**
+	 * Initialize the default value from the recipe of the selected design when the patterns are ready
+	 */
+	useEffect( () => {
+		if ( patterns.length === 0 || ! selectedDesign?.recipe ) {
```

Comment:
We have to wait until the patterns are ready as we will fetch the patterns from remote server instead of hard-coded theme in calypso.